homeassistant/components/overkiz/climate_entities/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py

```diff
@@ -0,0 +1,138 @@
+"""Support for Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    PRESET_BOOST,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_AUTO = "auto"
+PRESET_COMFORT1 = "comfort-1"
+PRESET_COMFORT2 = "comfort-2"
+PRESET_FROST_PROTECTION = "frost_protection"
+PRESET_PROG = "prog"
```

Comment:
It may be interesting to add all of them into a constant file into the climate_entities package.

Some are also use into 
https://github.com/home-assistant/core/blob/9c85d22bab333556efd4194fc74842b044412cfe/homeassistant/components/overkiz/climate_entities/atlantic_heat_recovery_ventilation.py#L24-L26

Do more Overkiz entities use those presets ?

Reply:
Bepending on which will be the first to be merged between this PR and #78659:
- OVERKIZ_TO_PRESET_MODE
- PRESET_MODE_TO_OVERKIZ
- OVERKIZ_TO_HVAC_MODE
- HVAC_MODE_TO_OVERKIZ

May share same constants

Reply:
@Quentame The constants are not exactly the same and are device dependant.  
Every devices supports it's own set of preset_mode and/or hvac mode.  
That's why it's set on the device file, used only on the device where it's defined.

In some case it could be the same, but there is often slight changes between devices...

Reply:
@Quentame We can indeed create constant for string such as auto, prog, or manual. 
As said @nyroDev, for constant like OVERKIZ_TO_PRESET_MODE they are not common to all the devices.

Reply:
@Quentame If you agree, I will create this constant file with another PR to not pollute this one.

Reply:
Ok for a future PR.

Yes, (some) devices will not use all entries of the Dict, but as you don't iterate on it and only use it as a key to value map: I don't think it will disturb anything.

Reply:
@Quentame but we do iterate on the dict? We have `HVAC_MODE_TO_OVERKIZ` which is the reverse of `OVERKIZ_TO_HVAC_MODE`, and we use this to define which HVAC modes are supported.

If the definitions are always the same, we could create a separate array for `_attr_hvac_modes`, but I don't think that we always have the same mappings.

Reply:
maybe the constants should be named differently on every devices?
Something like `HVAC_MODE_AEH_WITH_ADJUSTABLE_SET_POINT_TO_OVERKIZ`?
Will be very long but pretty explanatory that this constant is specific for this device.